lms/models/assignment.py

```diff
@@ -78,7 +78,7 @@ class Assignment(CreatedUpdatedMixin, Base):
     deep_linking_uuid: Mapped[str | None] = mapped_column(sa.Unicode, nullable=True)
     """UUID that identifies the deep linking that created this assignment."""
 
-    groupings: Mapped[list[Grouping]] = sa.orm.relationship(
+    groupings: DynamicMapped[Grouping] = sa.orm.relationship(
         secondary="assignment_grouping", viewonly=True, lazy="dynamic"
     )
```

Comment:
I note in [the docs](https://docs.sqlalchemy.org/en/20/orm/queryguide/relationships.html) that `lazy="dynamic"` is a deprecated feature. Is it planned to migrate to the new thing at some point?

Reply:
Assigned [this small note to myself](https://github.com/hypothesis/lms/issues/6445), I don't think it's that urgent to migrate away.

We'd keep this in mind to avoid adding new usages of it.